Feat ai feature

eitjank opened the pull request.

eitjank committed:
feat: separate free and premium insight dashboards

- Refactor the original `DashboardScreen` into a new `InsightDashboardScreen` which contains the premium features (Pie and Bar charts).
- Create a new, simpler `DashboardScreen` for the free tier, showing total expenses and a basic category breakdown.
- Include a placeholder card for a "Coming Soon" AI Monthly Forecast feature on the insight dashboard.

eitjank committed:
feat: integrate Gemini AI for financial advice

Alyas100 commented:
overall good, but i dont understand what the bar chart shows about, u can update that or explain to me here what it do, see below:
![WhatsApp Image 2025-10-07 at 9 49 54 PM](https://github.com/user-attachments/assets/5e05cb10-6583-4f47-8535-5a135447f185)

eitjank committed:
refactor: enhance pie chart interactivity and display

- Set the category name as center text on slice selection.
- Added a listener to handle chart value selection events.

Alyas100 reviewed (approved):
good
